### ISSUE TITLE
[bitnami/postgresql-ha] pgpool does not have persistence volume

### ISSUE THREAD

**Issue by ccmehk** [bitnami/charts#1631]
**Which chart**:

postgresql-ha v1.0.2

**Description**

**Steps to reproduce the issue:**

In order to support multiple users in pgpool, it is better to write the conf, such as pgpool.conf,  pool_hba.conf,  pool_passwd,  into persistence volume

**Describe the results you received:**

support read the config files in the /opt/bitnami/pgpool/conf from persistence volume

**Describe the results you expected:**

support read the config files in the /opt/bitnami/pgpool/conf from persistence volume

**Comment by beltran-rubo** [comment]
Hi, sorry for the delay on getting back to you. The main reason is to provide the ability to update the configuration every time you change any parameter, that is why those files do not persist in a volume.

In this specific case it looks like you will need to modify those files to add an user (if you do not use LDAP) so they are more similar to data than a config file.

My 2 cents, what it could be good for your case is to provide the ability to specify those files as configMaps (in the same way it is implemented for PostgreSQL conf files https://github.com/bitnami/charts/tree/master/bitnami/postgresql-ha#repmgrconf--postgresqlconf--pg_hbaconf-files-as-configmap). That way instead of adding the user directly to the deployed solution, you can store those files on your own and add them via options. That way the deployment will be updated with your own config files. Would that work for you?

**Comment by vincentbesanceney** [comment]
I think pool_passwd should be treated differently than other pgpool configuration files. You usually modify pool_passwd with the pg_md5 command, and you really want the users you add with this command to be persisted.

**Comment by juan131** [comment]
Hi @vincentbesanceney 

I would try to avoid persisting configuration, even if this configuration is created with commands that have to be run inside the container.

In this case (since they're credentials) I would use a secret to store all these data. For instance, we could add the possibility to add some secret with user/passwords information so you can add some script that auto-generates the **pool_passwd** file based on these credentials that are mounted in the container.

**Comment by ttinkr** [comment]
Was there any real solution to this? How should one manage multiple users between postgresql and pgpool without using LDAP?

**Comment by juan131** [comment]
Hi @ttinkr 

We don't have a solution for this yet. I proposed a solution some time ago (see https://github.com/bitnami/charts/issues/1631#issuecomment-565357598) but I was waiting for confirmation about my proposal. What do you think about it?

**Comment by ttinkr** [comment]
Hello @juan131!
A shared secret where the credentials are stored (and also the scripts to sync the content of the secret with pgpool_passwd and the postgresql user store) would be great, yes!

To be honest I am a little bit confused how people are using this in production then? Is everyone using only the one initial account to access the database? ...or is everyone using LDAP?

**Comment by juan131** [comment]
Hi @ttinkr 

I've been checking the changes to be implemented and theres's something extra to be done. 

We can create a secret with a list of users and passwords to be created. And then, use that information on some script to auto-generates the **pool_passwd** file:

```bash
read -r -a users <<< "$(tr ',;' ' ' <<< "${POSTGRESQL_USERS}")"
read -r -a passwords <<< "$(tr ',;' ' ' <<< "${POSTGRESQL_PASSWORDS}")"
if [[ "${#users[@]}" -ne "${#passwords[@]}" ]]; then
    error "Specify the same number of passwords on POSTGRESQL_PASSWORDS as the number of users on POSTGRESQL_USERS!"
    exit 1
fi
index=0
for user in "${users[@]}"; do
    ${passwords[$index]}
    pg_md5 -m --config-file=/opt/bitnami/pgpool/conf/pgpool.conf -u "$user" "${passwords[$index]}"
    index=$((index + 1 ))
done
```

That said, PostgreSQL nodes aren't aware of these new users. So we'll have to modify PostgreSQL nodes too with some script to create these users and grant permissions to them.

Does it make sense?

**Comment by ttinkr** [comment]
Makes totally sense, yes. Thanks for the quick solution to this!

**Comment by juan131** [comment]
Hi @ttinkr 

I collected all this information and created an internal task to add this feature in the chart. Thanks so much for your feedback!!

We'll keep you updated about our progress.

**Comment by tfellinger** [comment]
Hello @juan131 !

Sorry for being impatient, but do you have any timeline when this is going to be released?

Thanks,
Thomas

**Comment by ttinkr** [comment]
Hello @juan131 !

Sorry for being impatient, but do you have any timeline for releasing this?

Thanks,
Thomas

**Comment by juan131** [comment]
Hi @ttinkr 

I am afraid I can't provide you any exact ETA. I can tell you that the task is already assigned to one of our engineers, and it usually take a couple of weeks to be implemented once they're assigned.

Thanks for your consideration and patience.

**Comment by joancafom** [comment]
Hi @ttinkr !

We have already implemented changes in a PR #3238  and, once they are reviewed, they will be landed.

Thanks for your patience. Best regards.
